src/pages/AdminPage.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { addProduct, removeProduct } from "../store/products/actions";
+import { selectAllProducts } from "../store/products/selectors";
+import ProductCard from "../components/ProductCard";
+import "./AdminPage.scss";
+
+export default function AdminPage() {
+  const dispatch = useDispatch();
+  const [form, setForm] = useState({
+    name: "",
+    description: "",
+    url: "",
+    price: 0,
+    tags: [],
+    categories: [],
+  });
+  const [idToRemove, setIdToRemove] = useState("");
+  const products = useSelector(selectAllProducts);
+
+  const submitAddProductForm = (event) => {
+    event.preventDefault();
+
+    dispatch(
+      addProduct(
+        form.name,
+        form.description,
+        form.url,
+        Number(form.price),
+        form.tags,
+        form.categories
```

Comment:
This is really up to you to decide, but something that I like to do when a function gets many parameters is just wrap all the parameters in an object like:
```
const addProduct = ({ name, description, url, price, tags, categories }) => { ... }
// and then when used:
addProduct({
        name: form.name,
        description: form.description,
        url: form.url,
        price: Number(form.price),
        tags: form.tags,
        categories: form.categories
})
```

This makes it a bit more cumbersome to type when calling the function, but it makes the problem of remembering the order in which the parameters are passed obsolete, since its an object with keys so the order doesn't really matter. In this particular case it also allows you to do cooler and shorter things like:

```
dispatch(addProduct({ ...form, price: Number(form.price) })
```